src/pipes.cpp

```diff
@@ -0,0 +1,66 @@
+#include "pipes.h"
+
+const int Pipes::NUM_PIPES = 4;
+const float Pipes::PIPE_SCALE = 0.4;
+const int Pipes::PIPE_OFFSET_RANGE = 300;
+const float Pipes::SPEED_PER_SECOND = 500;
+
+Pipes::Pipes(int screenWidth, int screenHeight, int frameRate)
+{
+    bool res = pipeTexture.loadFromFile("./lib/pipe.png");
+    if(!res) throw("Cannot load pipe\n");
+
+    pipeSprites = new sf::Sprite[NUM_PIPES];
+    if (pipeSprites == nullptr) throw("Pipe array could not be created\n");
+
+    for(int i = 0 ; i < NUM_PIPES ; i++)
+    {
+        this->pipeSprites[i].setTexture(pipeTexture, true);
+        this->pipeSprites[i].setScale(PIPE_SCALE, PIPE_SCALE);
+
+        if(i % 2 == 1) this->pipeSprites[i].setRotation(180);
```

Comment:
This is an acceptable way to handle rotating every second pipe, but honestly I'd rather have seen a `Pipe` struct that owns it's own texture and has a rotation/position/size or at least parallel arrays that store each of these properties rather than the rotation be implicitly defines as every second pipe. This type of code leads to bad maintainability and is really really easy to overlook/forget and break